modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/test/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImplTest.java

```diff
@@ -372,6 +381,49 @@ private void _testExtractIndexableAttributes(
 				LocaleUtil.US));
 	}
 
+	private void _testFormWithRepeatableField(String indexType) {
+		Document document = _createDocument();
+
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+			SetUtil.fromArray(LocaleUtil.US), LocaleUtil.US);
+
+		DDMFormField ddmFormField = _createDDMFormField();
+
+		ddmFormField.setIndexType(indexType);
+		ddmFormField.setRepeatable(true);
+
+		ddmForm.addDDMFormField(ddmFormField);
+
+		List<LocalizedValue> localizedValues = ListUtil.fromArray(
+			new LocalizedValue() {
+				{
+					addString(LocaleUtil.US, "able");
+				}
+			},
+			new LocalizedValue() {
+				{
+					addString(LocaleUtil.US, "baker");
+				}
+			});
+
+		_ddmIndexer.addAttributes(
+			document, _createDDMStructure(ddmForm),
+			_createDDMFormValues(
+				ddmForm,
+				DDMFormValuesTestUtil.createDDMFormFieldValue(
+					_FIELD_NAME, localizedValues.get(0)),
+				DDMFormValuesTestUtil.createDDMFormFieldValue(
+					_FIELD_NAME, localizedValues.get(1))));
+
+		FieldValuesAssert.assertFieldValues(
+			_getSortableValues(
+				Collections.singletonMap(
+					"ddmFieldArray.ddmFieldValue" + indexType + "_en_US",
+					"[able, baker]")),
```

Comment:
@p-albuquerque one more question, shouldn't it be `["able", "baker"]` (array of strings) instead?

Reply:
Inside this assert method, the values extracted from the Document is transformed into a map and shaped as a unique string

![evidenceDoc](https://github.com/liferay-forms/liferay-portal/assets/79872134/e676c90c-9fb3-4f4e-b6ea-0aa91ed40608)